dao/src/main/java/org/thingsboard/server/dao/device/DeviceProfileServiceImpl.java

```diff
@@ -319,4 +350,37 @@ private DeviceProfileInfo toDeviceProfileInfo(DeviceProfile profile) {
                 profile.getDefaultDashboardId(), profile.getType(), profile.getTransportType());
     }
 
+    private void formatDeviceProfileCertificate(DeviceProfile deviceProfile, X509CertificateChainProvisionConfiguration x509Configuration) {
+        String certificateValue = formatCertificateValue(x509Configuration.getCertificateValue());
+        String cert = regexCertificateChain(certificateValue);
+        String sha3Hash = EncryptionUtil.getSha3Hash(cert);
+        DeviceProfileData deviceProfileData = deviceProfile.getProfileData();
+        x509Configuration.setCertificateValue(certificateValue);
+        deviceProfileData.setProvisionConfiguration(x509Configuration);
+        deviceProfile.setProfileData(deviceProfileData);
+        deviceProfile.setCertificateHash(sha3Hash);
+    }
+
+    private String regexCertificateChain(String chain) {
+        String regex = "-----BEGIN CERTIFICATE-----\\s*.*?\\s*-----END CERTIFICATE-----";
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(chain);
+        if (matcher.find()) {
+            return matcher.group(0);
+        }
+        return chain;
+    }
+
+    private String formatCertificateValue(String certificateValue) {
+        try {
+            CertificateFactory cf = CertificateFactory.getInstance("X.509");
+            ByteArrayInputStream inputStream = new ByteArrayInputStream(certificateValue.getBytes());
+            Certificate[] certificates = cf.generateCertificates(inputStream).toArray(new Certificate[0]);
+            if (certificates.length > 1) {
+                return EncryptionUtil.certTrimNewLinesForChainInDeviceProfile(certificateValue);
```

Comment:
If user provide chain of certificates, I don't understand how transport is going to map hash of multiple certificates to the hash of single certificate? In the transport certificates are processed one-by-one.